pandas/tests/scalar/timestamp/test_arithmetic.py

```diff
@@ -99,7 +99,7 @@ def test_rsub_dtscalars(self, tz_naive_fixture):
         if tz_naive_fixture is None:
             assert other.to_datetime64() - ts == td
         else:
-            msg = "subtraction must have"
+            msg = "Cannot subtract offset-naive and offset-aware datetimes."
```

Comment:
should have tests for the changed Timestamp behavior (i.e. that is tested in this file, not in the file above)

Reply:
ok, i can add tests here to cover subtraction behaviour of Timestamp. do you want coverage in both places? or is it enough just to test here?

Reply:
the scalar timestamp tests should be here.  the DatetimeArray/DatetimeIndex/Series[dt64] tests go in tests.arithmetic.test_datetime64

Reply:
sounds good :+1:

Reply:
@jbrockmendel any further comments here?

Reply:
Ok, i've moved the tests as requested